packages/more-candles/index.ts

```diff
@@ -0,0 +1,39 @@
+import { PluginInterface, Candle } from '@debut/types';
+
+interface MoreCandlesPluginInterface extends PluginInterface {
+    name: 'moreCandles';
+    api: {
+        getCandles: () => Candle[];
+    };
+}
+
+export type MoreCandlesOptions = {
+    amountOfCandles: number;
+};
+
+export interface MoreCandlesPluginAPI {
+    moreCandles: {
+        getCandles: () => Candle[];
+    };
+}
+
+export function moreCandlesPlugin(opts: MoreCandlesOptions): MoreCandlesPluginInterface {
+    let extendedCandles: Candle[] = [];
+
+    return {
+        name: 'moreCandles',
+        api: {
+            getCandles() {
+                return extendedCandles;
+            },
+        },
+
+        async onCandle(candle: Candle) {
+            extendedCandles.unshift(candle);
```

Comment:
shift operation is very slow, please copy fixed array size (CircularBuffer structure) from indicators library 
take a look here https://github.com/debut-js/Indicators/blob/master/src/providers/circular-buffer.ts

Reply:
i can help you if you need (bacouse current buffer does not have forEach method)